oximeter/db/src/oxql/query/mod.rs

```diff
@@ -362,6 +366,27 @@ impl Query {
     pub(crate) fn parsed_query(&self) -> &QueryNode {
         &self.parsed
     }
+
+    /// Insert filters after the `get`, or in the case of subqueries, recurse
+    /// down the tree and insert them after each get.
+    pub(crate) fn add_filters(&self, filters: Vec<Filter>) -> Self {
+        Self {
+            parsed: self.parsed.add_filters(filters),
+            end_time: self.end_time,
+        }
+    }
+}
+
+// TODO: move this to a more appropriate spot
+/// Just a helper for creating a UUID filter node concisely
+pub fn uuid_eq_filter(key: String, id: Uuid) -> Filter {
```

Comment:
Tiny QoL improvement, you could make this `fn uuid_eq_filter(key: impl ToString, id: Uuid) -> Filter`, and then call `key.to_string()` inside, instead of doing that at each callsite. Something like `AsRef<str>` would also work.